internal/resources/ibm/cloudant.go

```diff
@@ -0,0 +1,192 @@
+		Unit:            "GB",
+		MonthlyQuantity: q,
+		UnitMultiplier:  decimal.NewFromInt(1),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("ibm"),
+			Region:        strPtr(r.Region),
+			Service:       strPtr("cloudantnosqldb"),
+			ProductFamily: strPtr("service"),
+		},
+		PriceFilter: &schema.PriceFilter{
+			Unit: strPtr("GB_STORAGE_ACCRUED_PER_MONTH"),
+		},
+	}
+}
+
+func (r *Cloudant) cloudantReadsCostComponent() *schema.CostComponent {
+	c := convertCapacity(r.Capacity)
+
+	monthlyReads := c * 100
+
+	return &schema.CostComponent{
+		Name:            "Monthly Reads",
+		Unit:            "reads/second",
+		MonthlyQuantity: decimalPtr(decimal.NewFromInt(int64(monthlyReads))),
+		UnitMultiplier:  decimal.NewFromInt(1),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("ibm"),
+			Region:        strPtr(r.Region),
+			Service:       strPtr("cloudantnosqldb"),
+			ProductFamily: strPtr("service"),
+		},
+		PriceFilter: &schema.PriceFilter{
+			Unit: strPtr("READ_CAPACITY_ACCRUED_PER_MONTH"),
+		},
+	}
+}
+
+func (r *Cloudant) cloudantWritesCostComponent() *schema.CostComponent {
+	c := convertCapacity(r.Capacity)
+
+	monthlyWrites := c * 50
+
+	return &schema.CostComponent{
+		Name:            "Monthly Writes",
+		Unit:            "writes/second",
+		MonthlyQuantity: decimalPtr(decimal.NewFromInt(int64(monthlyWrites))),
+		UnitMultiplier:  decimal.NewFromInt(1),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("ibm"),
+			Region:        strPtr(r.Region),
+			Service:       strPtr("cloudantnosqldb"),
+			ProductFamily: strPtr("service"),
+		},
+		PriceFilter: &schema.PriceFilter{
+			Unit: strPtr("WRITE_CAPACITY_ACCRUED_PER_MONTH"),
+		},
+	}
+}
+
+func (r *Cloudant) cloudantGlobalQueriesCostComponent() *schema.CostComponent {
+	c := convertCapacity(r.Capacity)
+
+	monthlyGlobalQueries := c * 5
+
+	return &schema.CostComponent{
+		Name:            "Monthly Global Queries",
+		Unit:            "queries/second",
+		MonthlyQuantity: decimalPtr(decimal.NewFromInt(int64(monthlyGlobalQueries))),
+		UnitMultiplier:  decimal.NewFromInt(1),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr("ibm"),
+			Region:        strPtr(r.Region),
+			Service:       strPtr("cloudantnosqldb"),
+			ProductFamily: strPtr("service"),
+		},
+		PriceFilter: &schema.PriceFilter{
+			Unit: strPtr("GLOBAL_QUERY_CAPACITY_ACCRUED_PER_MONTH"),
+		},
+	}
+}
+
+// BuildResource builds a schema.Resource from a valid Cloudant struct.
+// This method is called after the resource is initialised by an IaC provider.
+// See providers folder for more information.
+func (r *Cloudant) BuildResource() *schema.Resource {
+	costComponents := []*schema.CostComponent{
+		r.cloudantReadsCostComponent(),
+		r.cloudantWritesCostComponent(),
+		r.cloudantGlobalQueriesCostComponent(),
+		r.cloudantFreeStorageCostComponent(),
+		r.cloudantStorageCostComponent(),
+	}
+
+	planName := cases.Title(language.Und).String(r.Plan)
```

Comment:
You need to handle the 3 plan types separately. Lite is free, standard has usage based costs and the dedicated one has a flat fee